### PR TITLE
Refactor wallet

### PR DESCRIPTION
**Pull request by andrewgordstewart**
The wallet is refactored for two purposes:

1. Currently, the wallet references some application states.
2. Having a `WalletEngine` that manages wallet state (like the game engine) makes it easy to transition to a "Challenge received" state, and update the view. ([Example of what we might do](https://github.com/magmo/rps-poc/compare/as-listen-for-challenges#diff-8a0206d2a470530ae09c70b55b9c41feR41))

Since the two wallet engines were almost identical, I chose to refactor them into one file. This is also a bit future-proof, for the case when there are more than one player.

### CODE REVIEW

File: src/wallet/redux/actions/external.ts [hunk 1]
```diff
@@ -18,11 +13,19 @@ export const FUNDING_FAILURE = 'WALLET.FUNDING.FAILURE';
 
 export const fundingRequest = (
   channelId: string,
-  state: WaitForFundingA | WaitForFundingB
+  myAddress: string,
```
Comment (andrewgordstewart) [inline src/wallet/redux/actions/external.ts]: this could (should?) be
```
channelId: string,
engineSetupArgs: WalletEngineSetupArgs
```

Comment (lalexgap) [reply src/wallet/redux/actions/external.ts]: I think it should stay this way, a funding request shouldn't need to know about engineArgs

File: src/redux/game/saga.ts [hunk 2]
```diff
@@ -75,10 +75,17 @@ function* sendState(state) {
...
-      yield put(walletActions.fundingRequest(state.channelId, state));
+      const { myAddress, opponentAddress, myBalance, opponentBalance } = state;
+      const playerIndex = (state.type === PlayerAStateType.WAIT_FOR_FUNDING) ? 0 : 1;
+      yield put(walletActions.fundingRequest(
+        state.channelId,
+        myAddress,
+        opponentAddress,
+        myBalance,
+        opponentBalance,
+        playerIndex,
```
[the remark this answers is not in the document]
Comment (andrewgordstewart) [reply src/redux/game/saga.ts]: The purpose was to remove the wallet's dependency on an RPS application state, though I suppose it wasn't necessary to unpack the arguments -- the wallet could define an interface defining what attributes the state needs.

[the remark this answers is not in the document]
Comment (tomclose) [reply src/redux/game/saga.ts]: I think we already have that interface - `State` from `fmg-core` has exactly the attributes common to all states, which should be all the wallet requires.

**Pull request merged**